Lib/tarfile.py

```diff
@@ -2567,7 +2568,19 @@ def chmod(self, tarinfo, targetpath):
         if tarinfo.mode is None:
             return
         try:
-            os.chmod(targetpath, tarinfo.mode)
+            try:
+                os.chmod(targetpath, tarinfo.mode)
+            except OSError as e:
+                if not(hasattr(errno, "EFTYPE") and e.errno == errno.EFTYPE):
```

Comment:
Maybe only retry chmod() when it's needed? Or is this test redundant with checking for EFTYPE?

```suggestion
                if not(hasattr(errno, "EFTYPE") and e.errno == errno.EFTYPE and (tarinfo.mode & stat.S_ISVTX)):
```

Reply:
Ok, no need to test for the bit in this case.